### PR TITLE
Bump `secp256k1` to 0.29

### PR DESCRIPTION
**Pull request by storopoli**
It would be really great if we could have a new `musig2` release with `0.0.12` that supports `rust-bitcoin` `0.32` that uses `secp256k1` `0.29`.
All test passes locally.
I took the liberty of also adding some clippy lint fixes in a separate commit.

Thank you for the awesome work!

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -16,8 +16,8 @@ hmac = { version = "0.12.1", default-features = false, features = [] }
 k256 = { version = "0.13.1", default-features = false, optional = true }
 once_cell = { version = "1.18.0", default-features = false }
 rand = { version = "0.8.5", optional = true, default-features = false, features = ["std_rng"] }
-secp = { version = "0.2.0", default-features = false }
-secp256k1 = { version = "0.28.0", optional = true, default-features = false }
+secp = { version = "0.3.0", default-features = false }
```
Comment (conduition) [inline Cargo.toml]: While we're at it, could you please change `secp`'s versions to `0.3`, to allow patches?

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "musig2"
-version = "0.0.11"
+version = "0.0.12"
```
Comment (conduition) [inline Cargo.toml]: Could you please remove the crate version bump here? I'll handle that part myself in a separate signed commit to `master` before publishing

**Review by conduition — COMMENTED** [review]
Thanks for this! i have a couple of requests, see below.

Also, could you please replace `cargo check` with `cargo clippy` in the `make check*` recipes? If we're going to enforce clippy's linting on this repo, we should also have an obvious way for devs to run clippy themselves against the various feature flag sets.

File: Cargo.toml (hunk 2 above)
Comment (storopoli) [reply Cargo.toml]: Sure done.

File: Cargo.toml (hunk 1 above)
Comment (storopoli) [reply Cargo.toml]: Yes, makes sense. Done

**Comment by storopoli** [comment]
> Also, could you please replace `cargo check` with `cargo clippy` in the `make check*` recipes? If we're going to enforce clippy's linting on this repo, we should also have an obvious way for devs to run clippy themselves against the various feature flag sets.

Done in 0831843b58d1c622aaeee0ac8f42e17c7cb8ca6f

File: Makefile [hunk 3]
```diff
@@ -7,22 +7,22 @@ check-default:
 
 # Checks the source code with all features enabled.
 check-mixed:
-	cargo check --all-features
-	cargo check --all-features --tests
+	cargo clippy --all-features
+	cargo clippy --all-features --tests
```
Comment (conduition) [inline Makefile:11]: There's one more `cargo check` that needs to be updated, L6 under the `check-default` recipe

File: src/bip340.rs [hunk 4]
```diff
@@ -409,10 +407,8 @@ mod tests {
             let verify_result = verify_single(pubkey, test_vec_signature, &record.message);
             match record.verification_result.as_str() {
                 "TRUE" => {
-                    verify_result.expect(&format!(
-                        "verification should pass for signature {} - {}",
-                        &record.signature, record.comment,
-                    ));
+                    verify_result.unwrap_or_else(|_| panic!("verification should pass for signature {} - {}",
+                        &record.signature, record.comment));
```
Comment (conduition) [inline src/bip340.rs]: I think you may need to run `cargo fmt` on this PR

Comment (conduition) [reply src/bip340.rs]: damn i really need to get off my ass, and set up github actions CI

EDIT: https://github.com/conduition/musig2/issues/6

Comment (storopoli) [reply src/bip340.rs]: Done

File: Makefile (hunk 3 above)
Comment (storopoli) [reply Makefile:11]: Done in 687b535

**Comment by storopoli** [comment]
Seeing #6, I already made the proper fix in 0c5140f so that `cargo doc --all-features` do not emit warnings anymore.

**Review by conduition — APPROVED** [review]

**Comment by conduition** [comment]
Included in `musig2@0.1.0`

**Comment by conduition** [comment]
Thank you very much for your help :smile:

**Pull request merged**